editor_drafts/draft-prorock-spice-cose-sd-cwt.md

```diff
@@ -54,287 +51,375 @@ document are to be interpreted as described in [@!RFC2119].
-implementation, binding can customize SD-CWT without compromising user
-privacy. Proportional use of holder binding balances security, privacy
-and flexibility for SD-CWT applications.
-
-### Counter Signatures
-
-Counter signatures allow an SD-CWT to be endorsed by additional entities
-beyond the original issuer. The counter signature is applied in the
-unprotected header, attesting to the validity of the primary SD-CWT
-signature over the protected claims.
-
-Counter signatures provide multiple benefits for SD-CWT:
-
-- Verification by multiple entities, preventing repudiation by any one
-  party.
-- Added integrity protection in case the original signing key is
-  compromised.
-- Allows separate signers for protected claims versus unprotected
-  disclosures.
-
-To utilize a counter signature, the primary SD-CWT is constructed and
-signed as normal. Then an additional signer computes the counter
-signature over the entire SD-CWT, including the original signature. This
-counter signature is placed in the unprotected header when transmitting
-the SD-CWT.
-
-Recipients validate the counter signature after verifying the primary
-signature, ensuring endorsements by all involved entities. Care should
-be taken to ensure robust trust in both signature authorities when
-relying on counter signatures.
-
-The CDDL fragment that represents an SD-CWT with an abbreviated counter 
-signature is below:
+The verifier MUST compute the hash of the sd-cwt-claim-pair, in order to match the disclosed value to redacted claims in the SD-CWT.
 
-```
-SD-CWT = [
-    protected,
-    unprotected: {
-      ? disclosures: [* claim-pair] / nil
-      COSE_Countersignature0: bstr
-    },
-    payload : bstr / nil,
-    signature : bstr,
-]
-```
+To verify an SD-CWT, the recipient extracts the protected CWT claims from the payload. 
+These CWT claims contain hash digests of the original claim values combined with unique random salts.
 
-## Data Structures
+By performing these steps, the recipient can cryptographically verify the integrity of the protected claims and verify they have not been tampered with or substituted after issuance by the trusted issuer. 
+The disclosures provide the plaintext claim values for utilization by the recipient.
 
-TBD - Describe common data structures in CDDL
+The algorithm for transorming the CWT Claimset mirrors the algorithm defined in SD-JWT.
+The primary differences are that CBOR maps have replaced JSON Objects, and that CBOR labels (integers) have replaced the strings used by SD-JWT which are `_sd` and `...`.
+This specification uses the term `validated_presented_sd_claims` to refer to the final CBOR map which is produced by subsituting disclosed values which are presented, and removing labels marked for redaction.
+
+As described in SD-JWT, if there remain unused SD-CWT disclosures at the end of this procedure the SD-CWT MUST be considered invalid, as if the siganture had failed to verify.
```

Comment:
I would find it much clearer if the algorithm for transforming the CWT Claimset were described plainly, without reference to SD-JWT.

The audience of this document will likely include people who aren't deeply familiar with SD-JWT, and for them the reference to SD-JWT will likely be confusing rather than enlightening.

Reply:
We'll end up repeating a lot of the sd-jwt draft text. I'm fine doing that, and I think it's the right thing to do eventually, but I didn't want to do it in this PR.